src/core/instructions.jl

```diff
@@ -200,10 +200,10 @@ Base.getindex(iter::PhiIncomingSet, i) =
     tuple(Value(API.LLVMGetIncomingValue(ref(iter.phi), Cuint(i-1))),
                 BasicBlock(API.LLVMGetIncomingBlock(ref(iter.phi), Cuint(i-1))))
 
-function Base.append!(iter::PhiIncomingSet, args::Vector{Tuple{Value, BasicBlock}})
+function Base.append!(iter::PhiIncomingSet, args::Vector{Tuple{V, BasicBlock}} where V <: Value)
```

Comment:
perhaps a `push!(iter, (Value, BasicBlock)` could be added?

Reply:
Yeah that seems sensible, not sure why I didn't initially. `append!` can use `push!` then.